libsql/internal/hrana/batch_result.go

```diff
@@ -1,6 +1,49 @@
 package hrana
 
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+)
+
 type BatchResult struct {
-	StepResults []*StmtResult `json:"step_results"`
-	StepErrors  []*Error      `json:"step_errors"`
+	StepResults      []*StmtResult `json:"step_results"`
+	StepErrors       []*Error      `json:"step_errors"`
+	ReplicationIndex *uint64       `json:"replication_index"`
```

Comment:
go nood question: why does `ReplicationIndex` have to be a pointer here? And where does it point to?

Reply:
to represent the case when we don't have ReplicationIndex - I guess